tests/Doctrine/Tests/Common/Proxy/AbstractProxyFactoryTest.php

```diff
@@ -215,6 +216,170 @@ public function testMissingPrimaryKeyValue()
+            ->willReturnCallback(function() use ($proxyFile) {
+                file_put_contents($proxyFile, '<?php class MyObject1 {} ');
+            });
+
+        /** @var MockObject&AbstractProxyFactory $proxyFactory */
+        $proxyFactory = $this->getMockForAbstractClass(
+            AbstractProxyFactory::class,
+            [$proxyGenerator, $metadataFactory, AbstractProxyFactory::AUTOGENERATE_FILE_NOT_EXISTS_OR_CHANGED]
+        );
+
+        $proxyFactory
+            ->expects($this->any())
+            ->method('createProxyDefinition')
+            ->will($this->returnValue($definition));
+
+        $generatedProxy = $proxyFactory->getProxy('Class', ['id' => 1]);
+
+        self::assertInstanceOf('MyObject1', $generatedProxy);
+    }
+
+    public function testGetProxyFileWhenProxyIsOlderThanSource()
+    {
+        $proxyFile = tempnam(sys_get_temp_dir(), 'proxy');
+        file_put_contents($proxyFile, '<?php class MyObject2 {} ');
+        sleep(1);
+        $sourceFile = tempnam(sys_get_temp_dir(), 'source');
+
+        $metadata        = $this->createMock(ClassMetadata::class);
+        $definition      = new ProxyDefinition('MyObject2', [], [], null, null);
+        $proxyGenerator  = $this->createMock(ProxyGenerator::class);
+        $metadataFactory = $this->createMock(ClassMetadataFactory::class);
+        $reflection      = $this->createMock(ReflectionClass::class);
+
+        $metadataFactory
+            ->expects($this->once())
+            ->method('getMetadataFor')
+            ->willReturn($metadata);
+
+        $metadata
+            ->expects($this->once())
+            ->method('getName')
+            ->willReturn('MyObject2');
+
+        $metadata
+            ->expects($this->once())
+            ->method('getReflectionClass')
+            ->willReturn($reflection);
+
+        $reflection
+            ->expects($this->once())
+            ->method('getFileName')
+            ->willReturn($sourceFile);
+
+        $proxyGenerator
+            ->expects($this->once())
+            ->method('getProxyFileName')
+            ->willReturn($proxyFile);
+
+        $proxyGenerator
+            ->expects($this->once())
+            ->method('generateProxyClass');
+
+        /** @var MockObject&AbstractProxyFactory $proxyFactory */
+        $proxyFactory = $this->getMockForAbstractClass(
+            AbstractProxyFactory::class,
+            [$proxyGenerator, $metadataFactory, AbstractProxyFactory::AUTOGENERATE_FILE_NOT_EXISTS_OR_CHANGED]
+        );
+
+        $proxyFactory
+            ->expects($this->any())
+            ->method('createProxyDefinition')
+            ->will($this->returnValue($definition));
+
+        $generatedProxy = $proxyFactory->getProxy('Class', ['id' => 1]);
+
+        self::assertInstanceOf('MyObject2', $generatedProxy);
+    }
+
+    public function testGetProxyFileWhenProxyIsNewerThanSource()
+    {
+        $sourceFile = tempnam(sys_get_temp_dir(), 'source');
+        sleep(1);
```

Comment:
IMO it's not such a big deal. If it becomes one in the future, we should consider introducing something like `lcobucci/clock`.